### PR TITLE
Merge with PostgreSQL STABLE 9.4.26.

### PR DESCRIPTION
**Pull request by guofengrichard**
This PR merges PostgreSQL commits from REL9_4_24 to REL9_4_26.

Some notable changes include but are not limited to:

* Allow repalloc() to give back space when a large chunk is reduced in
  size
* Allow configure --with-python to succeed when only python3 or only
  python2 can be found
* Fix possible crash with a SubPlan (sub-SELECT) within a multi-row
  VALUES list
* Improve error reporting in to_date() and to_timestamp()
* Improve performance of hash joins with very large inner relations

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by interma** [comment]
The pipeline is nearly passed:
https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-merge-iteration_REL9_4_STABLE_9.4.26

The failed jobs are caused by ccp env issue, instead of failing the test.

**Comment by asimrp** [comment]
Hey @guofengrichard thank you for merging the latest 9_4_STABLE!

Typically, upstream merges like this one are brought in with a single merge commit, the steps for creating an iteration branch and a PR are outlined here: https://github.com/greenplum-db/gpdb-postgres-merge/wiki/How-to-create-merge-iteration-branch.  Can we adjust this PR accordingly?

**Comment by asimrp** [comment]
Edit: also thank you @interma for your help in conflict resolution and getting tests green!

**Comment by asimrp** [comment]
> Typically, upstream merges like this one are brought in with a single merge commit, the steps for creating an iteration branch and a PR are outlined here: https://github.com/greenplum-db/gpdb-postgres-merge/wiki/How-to-create-merge-iteration-branch. Can we adjust this PR accordingly?

Apologies, I take that back! The empty commit message and Github UI baffled me.  After fetching the PR in my git repo, I realized that this is a merge commit.

**Comment by asimrp** [comment]
Nit: some details can be added to the commit message, such as:
> This PR merges PostgreSQL commits from REL9_4_24 to REL9_4_26.

and notable bugs or particularly tricky conflict resolutions, if any.

**Comment by adam8157** [comment]
> > Typically, upstream merges like this one are brought in with a single merge commit, the steps for creating an iteration branch and a PR are outlined here: https://github.com/greenplum-db/gpdb-postgres-merge/wiki/How-to-create-merge-iteration-branch. Can we adjust this PR accordingly?
> 
> Apologies, I take that back! The empty commit message and Github UI baffled me. After fetching the PR in my git repo, I realized that this is a merge commit.

Haha, it fooled me too. GitHub UI shows the commits equally, but they are at different levels, like:

```
Merge commit
    Upstream commit 0
    Upstream commit 1
    Upstream commit 2
    ...
```

**Comment by guofengrichard** [comment]
> Nit: some details can be added to the commit message, such as:
> 
> > This PR merges PostgreSQL commits from REL9_4_24 to REL9_4_26.
> 
> and notable bugs or particularly tricky conflict resolutions, if any.

Thanks for the suggestion. Have updated the commit message.

**Comment by ashwinstar** [comment]
> Allow configure --with-python to succeed when only python3 or only
python2 can be found

Please can you point to specific change / commit for this (lazily asking instead of trying to locate it). Greenplum 6 doesn't support python 3 directly, so wish to avoid any behavior difference due to this on STABLE branch.

**Comment by guofengrichard** [comment]
> > Allow configure --with-python to succeed when only python3 or only
> > python2 can be found
> 
> Please can you point to specific change / commit for this (lazily asking instead of trying to locate it). Greenplum 6 doesn't support python 3 directly, so wish to avoid any behavior difference due to this on STABLE branch.

Here is the commit: 96db9d5e2b

File: config/python.m4 [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # "python2" and "python3", in which case it's reasonable to prefer the
 # newer version.
 AC_DEFUN([PGAC_PATH_PYTHON],
-[PGAC_PATH_PROGS(PYTHON, [python python2])
+[PGAC_PATH_PROGS(PYTHON, [python python3 python2])
```
Comment (ashwinstar) [inline config/python.m4:19]: we shouldn't be including `python3` for GPDB6 as that would be behavior change. Previous code change from this upstream commit was manually pulled in to GDPB6 except python3 refer 9f94b9771547f85313734de3c388a02d0222373e

**Comment by ashwinstar** [comment]
Was curious to look into what kind of fixes this merge brings and how many are super critical in nature from GPDB perspective. Following commits I found are really relevant and critical to GPDB (others are low risk and good to have category)

- https://github.com/greenplum-db/gpdb/pull/11784/commits/fbe897134801504a3f1240d61cb78a69bc0850a5
- https://github.com/greenplum-db/gpdb/pull/11784/commits/5f73d7294d9fbaa36a0167e7aa6fa693c9c724e6
- https://github.com/greenplum-db/gpdb/pull/11784/commits/29e47f83caf38ca7967655a4a71d64f30a6461b5 (updates timezome data files to 2019c)
- https://github.com/greenplum-db/gpdb/pull/11784/commits/8a17afe84be6fefe76d0d2f4d26c5ee075e64487
- https://github.com/greenplum-db/gpdb/pull/11784/commits/080cf32d22317591cd8891c797a4d925cf1dbda2
- https://github.com/greenplum-db/gpdb/pull/11784/commits/f88f7206edbfe9d06655e3885219d812b0362251
- https://github.com/greenplum-db/gpdb/pull/11784/commits/65da6dd1d306dd679b92770b628d44ed6a08db3b (fix view dumping)
- https://github.com/greenplum-db/gpdb/pull/11784/commits/ba1dfbe22d3002ff56933ee6ebd26b9bc9be3d86

File: config/python.m4 (hunk 1 above)
Comment (guofengrichard) [reply config/python.m4]: Thanks for pointing this out. Have removed `python3`.

**Comment by adam8157** [comment]
Rebased on latest 6X_STABLE

**Comment by adam8157** [comment]
FYI, if you see "This branch cannot be rebased due to conflicts" on this page, it's not true, as of now.

**Pull request merged**